src/components/Modals/AddModal.vue

```diff
@@ -1,62 +1,354 @@
 <script setup lang="ts">
-import { useModal } from '@/composables/modal';
-import { ref } from 'vue';
+import { ref, reactive, computed, watch } from 'vue';
 import { useDisplay } from 'vuetify'
+import { useI18n } from 'vue-i18n';
+import { useModal } from '@/composables/modal';
+import { usePreferences } from '@/composables/api/preferences';
+import { useCategories, useTags, useTorrents } from '@/composables/api/torrents';
+import { useValidator } from '@/composables/validator';
+import { DEFAULT_DEST_FOLDER_PATH } from '@/constants';
+import { 
+  mdiClose, 
+  mdiCloudUpload, 
+  mdiPaperclip,
+  mdiLink,
+  mdiTag,
+  mdiLabel,
+  mdiFolder,
+} from '@mdi/js'
 
-// TODO : implement the component based on the one already present on the legacy
+import type { AddTorrentsPayload } from '@/types/vuetorrent/payloads/AddTorrentsPayload';
+import { AppPreferences } from '@/enums/qbit';
 
 const props = defineProps<{
   guid: string
 }>()
 
 const { dialog } = useModal(props.guid)
 const display = useDisplay()
+const { torrentFilesRule } = useValidator()
+const { categories: availableCategories } = useCategories()
+const { tags: availableTags } = useTags()
+const { data: preferences, isLoading: isPreferencesLoading } = usePreferences()
+const { t } = useI18n()
+const { addTorrents } = useTorrents()
+
+const addTorrentButton = ref<HTMLButtonElement>()
+const form = ref<HTMLFormElement>()
+
+const isFormValid = ref(false)
+const isDropAreaVisible = ref(false)
+
+const params = reactive<AddTorrentsPayload>({
+  urls: '',
+  files: [],
+  tags: [],
+  category: undefined,
+  directory: DEFAULT_DEST_FOLDER_PATH,
```

Comment:
for this one, it would be awesome to update the value when the category is changed

Reply:
mb, haven't seen the watch ^^